.azure-pipelines/windows-powershell.yml

```diff
@@ -36,46 +39,40 @@ jobs:
 - job: Analyze
   displayName: Analyze
   dependsOn: Build
-  condition: and(eq(variables.IsGenerateBased, false), succeeded())
-  timeoutInMinutes: 90
+  condition: succeeded()
+  timeoutInMinutes: ${{ variables.AnalysisTimeoutInMinutes }}
+  strategy:
+    matrix:
+      windows:
+        OSName: ${{ variables.WindowsName }}
+        ImageName: ${{ variables.WindowsImage }}
   pool:
-    vmImage: ${{ variables.WindowsImage }}
+    vmImage: $(ImageName)
 
   steps:
   - template: util/analyze-steps.yml
     parameters:
-      osName: ${{ variables.WindowsName }}
+      osName: $(OSName)
       configuration: ${{ variables.Configuration }}
-  - task: PowerShell@2
-    displayName: 'Cleanup Build'
-    inputs:
-      targetType: filePath
-      filePath: tools/CleanupBuild.ps1
-      pwsh: true
-  - task: NuGetCommand@2
-    displayName: 'Download BinSkim'
-    inputs:
-      command: custom
-      arguments: 'install Microsoft.CodeAnalysis.BinSkim -OutputDirectory $(System.DefaultWorkingDirectory)/tools/SecurityTools -Version 1.7.0'
-  - task: PowerShell@2
-    displayName: 'Run BinSkim'
-    inputs:
-      targetType: filePath
-      filePath: tools/SecurityTools/RunBinSkim.ps1
-      pwsh: true
+      testFramework: ${{ variables.TestFramework }}
 
 - job: Test
   displayName: Test
   dependsOn: Build
-  condition: and(eq(variables.IsGenerateBased, false), succeeded())
+  condition: succeeded()
```

Comment:
Windows Build has been disabled in generation CI. So I do not think we should change the condition here. Besides, I think pipeline  related changes should be verified both in main and generation branch before merging.

Reply:
Updated.